ObjectiveGit/GTRepository+Merging.m

```diff
@@ -0,0 +1,215 @@
+//
+//  GTRepository+Merging.m
+//  ObjectiveGitFramework
+//
+//  Created by Piet Brauer on 02/03/16.
+//  Copyright © 2016 GitHub, Inc. All rights reserved.
+//
+
+#import "GTRepository+Merging.h"
+#import "GTOID.h"
+#import "NSError+Git.h"
+#import "git2/errors.h"
+#import "GTCommit.h"
+#import "GTReference.h"
+#import "GTRepository+Committing.h"
+#import "GTRepository+Pull.h"
+#import "GTTree.h"
+#import "GTIndex.h"
+#import "GTIndexEntry.h"
+
+typedef void (^GTRemoteFetchTransferProgressBlock)(const git_transfer_progress *stats, BOOL *stop);
+
+@implementation GTRepository (Merging)
+
+typedef void (^GTRepositoryEnumerateMergeHeadEntryBlock)(GTCommit *entry, BOOL *stop);
+
+typedef struct {
+	__unsafe_unretained GTRepository *repository;
+	__unsafe_unretained GTRepositoryEnumerateMergeHeadEntryBlock enumerationBlock;
+} GTEnumerateMergeHeadEntriesPayload;
+
+int GTMergeHeadEntriesCallback(const git_oid *oid, void *payload) {
+	GTEnumerateMergeHeadEntriesPayload *entriesPayload = payload;
+
+	GTRepository *repository = entriesPayload->repository;
+	GTRepositoryEnumerateMergeHeadEntryBlock enumerationBlock = entriesPayload->enumerationBlock;
+
+	GTCommit *commit = [repository lookUpObjectByOID:[GTOID oidWithGitOid:oid] objectType:GTObjectTypeCommit error:NULL];
```

Comment:
Hm, you basically take the results of this method and put it into the parents of the merge commit. I mean it's only a matter of another map so I omitted that. But we totally could switch to `GTOID`s as this would map 1:1 to the libgit2 functionality.